apps/trading/client-pages/market/market.tsx

```diff
@@ -31,17 +32,54 @@ export interface SingleMarketData extends SingleMarketFieldsFragment {
   data: MarketData;
 }
 
-export const Market = () => {
+const TitleUpdater = ({
+  marketId,
+  marketName,
+  decimalPlaces,
+}: {
+  marketId?: string;
+  marketName?: string;
+  decimalPlaces?: number;
+}) => {
+  const pageTitle = usePageTitleStore((store) => store.pageTitle);
+  const updateTitle = usePageTitleStore((store) => store.updateTitle);
+  const { data: marketData } = useThrottledDataProvider<
+    MarketData,
+    MarketDataUpdateFieldsFragment
+  >(
+    {
+      dataProvider: marketDataProvider,
+      variables: useMemo(() => ({ marketId }), [marketId]),
```

Comment:
suggestion (non blocking): It would be good if the data providers memoized the query variables themselves

Reply:
Good point. Addressed in below PR
https://github.com/vegaprotocol/frontend-monorepo/pull/2693